File: payment-service-node/src/clients/riskClient.js
const axios = require('axios');
const config = require('../config');

const client = axios.create({
  baseURL: config.services.riskUrl,
  timeout: 5000,
});

/**
 * Check transaction risk via the risk-service.
 * @param {string} fromAccount - Source account ID
 * @param {string} toAccount - Destination account ID
 * @param {number} amount - Amount in cents
 * @returns {Promise<boolean>} true if approved, false if rejected
 */
async function checkRisk(fromAccount, toAccount, amount) {
  try {
    const response = await client.post('/api/v1/risk/check', null, {
      params: {
        from_account: fromAccount,
        to_account: toAccount,
        amount,
      },
    });
    return response.data === true;
  } catch (err) {
    if (err.response && err.response.status === 400) {
      return false;
    }
    throw err;
  }
}

module.exports = { checkRisk };

File: payment-service/src/main/java/com/payflow/payment/client/RiskClient.java
package com.payflow.payment.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(name = "risk-service", url = "${services.risk.url}")
public interface RiskClient {

    @PostMapping("/api/v1/risk/check")
    boolean checkRisk(
            @RequestParam("from_account") String fromAccount,
            @RequestParam("to_account") String toAccount,
            @RequestParam("amount") Long amount);
}
